scipy/integrate/_quadrature.py

```diff
@@ -662,6 +664,254 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
+    relationship between each point and the two adjacent points.
+
+    Parameters
+    ----------
+    y : array_like
+        Values to integrate.
+    x : array_like, optional
+        The coordinate to integrate along. Must be monotonically increasing.
+        If None (default), use spacing `dx` between consecutive elements in `y`.
+    dx : scalar | array_like, optional
+        Spacing between elements of `y`. Only used if `x` is None. Can either be
+        a float, or an array with the same shape as `y`, but size 1 along `axis`.
+    axis : int, optional
+        Specifies the axis to integrate along. Default is -1 (last axis).
+    initial : scalar | array_like, optional
+        If given, insert this value at the beginning of the returned result,
+        and add it to the rest of the result. Default is None, which means no
+        value at ``x[0]`` is returned and `res` has one element less than `y`
+        along the axis of integration. Can either be a float, or an array with
+        the same shape as `y`, but size 1 along `axis`.
+
+    Returns
+    -------
+    res : ndarray
+        The result of cumulative integration of `y` along `axis`.
+        If `initial` is None, the shape is such that the axis of integration
+        has one less value than `y`. If `initial` is given, the shape is equal
+        to that of `y`.
+
+    See Also
+    --------
+    numpy.cumsum, numpy.cumprod
+    cumulative_trapezoid : cumulative integration using the composite trapezoidal rule
+    simpson : integrator for sampled data using the Composite Simpson's Rule
+    quad : adaptive quadrature using QUADPACK
+    romberg : adaptive Romberg quadrature
+    quadrature : adaptive Gaussian quadrature
+    fixed_quad : fixed-order Gaussian quadrature
+    dblquad : double integrals
+    tplquad : triple integrals
+    romb : integrators for sampled data
+    ode : ODE integrators
+    odeint : ODE integrators
+
+    Notes
+    -----
+
+    .. versionadded:: 1.11.0
+
+    For samples that are equally spaced the result is exact if the function
+    is a polynomial of order 3 or less. If the samples are not equally spaced,
+    then the result is exact only if the function is a polynomial of order 2
+    or less.
+
+    The composite Simpson's 1/3 method is outlined in [1]. A quadratic relationship
+    is assumed between any 3 adjacent points (2 subintervals, :math:`h_{1}` and
+    :math:`h_{2}`). The integral is estimated over each subinterval and cumulatively
+    summed.
+
+    References
+    ----------
+    .. [1] Wikipedia page: https://en.wikipedia.org/wiki/Simpson's_rule
+
+    Examples
+    --------
+    >>> from scipy import integrate
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+
+    >>> x = np.linspace(-2, 2, num=20)
+    >>> y = x
+    >>> y_int = integrate.cumulative_simpson(y, x, initial=0)
+    >>> plt.plot(x, y_int, 'ro', x, y[0] + 0.5 * x**2, 'b-')
+    >>> plt.show()
+
+    """
+
+    y = np.asarray(y)
```

Comment:
```suggestion
    """
    y = np.asarray(y)
```

Reply:
Addressed in [be6e1d7](https://github.com/scipy/scipy/pull/18151/commits/be6e1d75e7863d46c297ad24a9929392873dcf3a)